src/solvers/refinement/string_refinement.cpp

```diff
@@ -96,38 +96,40 @@ static std::vector<exprt> instantiate_not_contains(
   const string_constraint_generatort &generator);
 
 static exprt get_array(
-  std::function<exprt(const exprt &)> super_get,
+  const std::function<exprt(const exprt &)> &super_get,
   const exprt &arr);
 
 /// Convert exprt to a specific type. Throw bad_cast if conversion
 /// cannot be performed
 /// Generic case doesn't exist, specialize for different types accordingly
 /// TODO: this should go to util
-template<typename T>
-optionalt<T> expr_cast(const exprt &);
 
-template<>
-optionalt<mp_integer> expr_cast<mp_integer>(const exprt &expr)
+// Tag dispatching struct
+template <typename Wrapped>
+struct type_wrappert {};
+
+optionalt<mp_integer> expr_cast(
+  const exprt &expr, type_wrappert<mp_integer>)
 {
   mp_integer out;
   if(to_integer(expr, out))
     return { };
   return out;
 }
 
-template<>
-optionalt<std::size_t> expr_cast<std::size_t>(const exprt &expr)
+optionalt<std::size_t> expr_cast(
+  const exprt &expr, type_wrappert<std::size_t>)
```

Comment:
That change should be reverted. It is a workaround for a bug in CLion linter. [This template is specialized correctly].(http://en.cppreference.com/w/cpp/language/template_specialization).
Should erroneous CLion warnings be taken into consideration, version based on class/struct templates should be tried first, before using this. Verbosity should be taken away from the user at all costs.

Reply:
Added your changes now.